### PR TITLE
Support for reading pretrained YoloX models with old blocks

### PR DESCRIPTION
**Pull request by shaydeci**
- Old Focus block weights reshaped when reading ckpt to 6x6 conversation weights.
- Anchors are fixed to old COCO Anchors for compatibility and will be removed in the future.

### CODE REVIEW

File: src/super_gradients/training/utils/checkpoint_utils.py [hunk 1]
```diff
@@ -222,6 +225,20 @@ def __init__(self, desc):
         self.message = "Missing pretrained wights: " + desc
         super().__init__(self.message)
 
+def _yolox_ckpt_solver(ckpt_key, ckpt_val, model_key, model_val):
+    """
+    Helper method for reshaping old pretrained checkpoint's focus weights to 6x6 conv weights.
+    """
+    if ckpt_key == 'module._backbone._modules_list.0.conv.conv.weight' and model_key == '_backbone._modules_list.0.conv.weight':
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/utils/checkpoint_utils.py:232]: They aren't exactly the same.

File: src/super_gradients/training/utils/checkpoint_utils.py [hunk 2]
```diff
@@ -132,20 +132,23 @@ def read_ckpt_state_dict(ckpt_path: str, device="cpu"):
...
-def adapt_state_dict_to_fit_model_layer_names(model_state_dict: dict, source_ckpt: dict, exclude: list = []):
+def adapt_state_dict_to_fit_model_layer_names(model_state_dict: dict, source_ckpt: dict, exclude: list = [], solver: callable=None):
     """
     Given a model state dict and source checkpoints, the method tries to correct the keys in the model_state_dict to fit
     the ckpt in order to properly load the weights into the model. If unsuccessful - returns None
         :param model_state_dict:               the model state_dict
         :param source_ckpt:                         checkpoint dict
-        :exclude                  optional list for excluded layers
+        :param exclude                  optional list for excluded layers
+        :param solver:                  callable that returns a desired weight for (ckpt_key, ckpt_val, model_key, model_val)
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/utils/checkpoint_utils.py]: Updated in https://github.com/Deci-AI/super-gradients/pull/270/commits/9b4fea509d83b5e260a303b620fc13359d3a0835

File: src/super_gradients/training/utils/checkpoint_utils.py [hunk 3]
```diff
@@ -222,6 +225,20 @@ def __init__(self, desc):
         self.message = "Missing pretrained wights: " + desc
         super().__init__(self.message)
 
+def _yolox_ckpt_solver(ckpt_key, ckpt_val, model_key, model_val):
+    """
+    Helper method for reshaping old pretrained checkpoint's focus weights to 6x6 conv weights.
+    """
+    if ckpt_key == 'module._backbone._modules_list.0.conv.conv.weight' and model_key == '_backbone._modules_list.0.conv.weight':
+        model_val.data[:, :, ::2, ::2] = ckpt_val.data[:, :3]
+        model_val.data[:, :, 1::2, ::2] = ckpt_val.data[:, 3:6]
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/utils/checkpoint_utils.py:236]: updated in https://github.com/Deci-AI/super-gradients/pull/270/commits/9b4fea509d83b5e260a303b620fc13359d3a0835

**Review by yurkovak — APPROVED** [review]
LGTM

**Review by ofrimasad — APPROVED** [review]
LGTM
